llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -10272,6 +10280,8 @@ void llvm::findValuesAffectedByCondition(
                                                            m_Value()))) {
       // Handle patterns that computeKnownFPClass() support.
       AddAffected(A);
+    } else if (!IsAssume && match(V, m_Not(m_Value(X)))) {
```

Comment:
Can we drop the !IsAssume here and instead drop the m_Not special case for assumes above?

Reply:
to be able to do that `isEphemeralValueOf `need to be updated to handle that.
example of test that will fail as the assume is optimized away.
https://github.com/llvm/llvm-project/blob/3d140004c70e2bc79416825e43207e8b711c56d9/llvm/test/Transforms/GVN/assume.ll#L48-L61

Maybe all instructions with boolean results building up the condition for the assume shall be Ephemeral.

Reply:
Have some follow up changes that update `isEphemeralValueOf`  here https://github.com/llvm/llvm-project/compare/main...andjo403:llvm-project:assumeNotCond but probably better as multiple PRs